### ISSUE TITLE
Leverage awsdk to add facts 

### ISSUE THREAD

**Issue by MarsuperMammal** [puppetlabs/puppetlabs-aws#119]
I'm going to be looking into this myself, likely today, but it will be an early step into writing custom facts that do something other than call system binaries.

The ability to set tags inside this module is the most interesting piece to me. Using tags as a staging point for associations between AWS resources allows grouping and treating resources as end services for management without requiring the extremely tight, and somewhat restrictive coupling added by CloudFormation or similar tools.

I'm assuming it shouldn't be terribly hard to pull tags for a resource out of AWS as a fact. It might also be useful to push those tags into the puppet resource tags property for any resource that manages the creation of an element in AWS.

Am I off base? Does this seem useful to anyone else? Is this trivial, and I'm just not particularly smart? All valid possibilites.

**Comment by garethr** [comment]
HI @MarsuperMammal. Additional EC2 facts definitely sound of interest. I presume you've already seen the existing core facts http://docs.puppetlabs.com/facter/latest/core_facts.html#ec2ec2-instance-data

It's probably worth kicking off a conversation about where those facts should live on the puppet-dev mailing list I think. That way the conversation will hopefully involve some of the people working on facter as well as those interested in this module.

**Comment by MarsuperMammal** [comment]
Yeah, I think the facts that come in to core facter are purely the data returned by dumping the instance metadata. There's a number of things about them that are less than ideal in terms of usability, and a lot of information available from API endpoints in AWS not covered in the metadata. I'll start a conversation in puppet-dev
